prov/rxm/src/rxm_ep.c

```diff
@@ -1985,10 +2011,36 @@ static int rxm_ep_msg_res_open(struct rxm_ep *rxm_ep)
 	return ret;
 }
 
+static void rxm_ep_sar_init(struct rxm_ep *rxm_ep)
+{
+	size_t param;
+
+	if (!fi_param_get_size_t(&rxm_prov, "sar_limit", &param)) {
+		if (param < rxm_info.tx_attr->inject_size) {
+			FI_WARN(&rxm_prov, FI_LOG_CORE,
+				"Requested SAR limit (%zd) less than inject size (%zd). "
+				"SAR protocol won't be used. Messages of size <= (>) inject "
+				"size would would be transmitted via eager (rendezvous) "
+				"protocol.\n", param, rxm_info.tx_attr->inject_size);
+		} else {
+			rxm_ep->sar.limit = param;
+			rxm_ep->sar.segs_cnt_limit =
+				rxm_ep_sar_calc_segs_cnt(rxm_ep, rxm_ep->sar.limit);
+		}
+	} else {
+		rxm_ep->sar.segs_cnt_limit = rxm_ep->msg_info->tx_attr->size;
+		rxm_ep->sar.limit = rxm_ep->sar.segs_cnt_limit * rxm_info.tx_attr->inject_size;
+		if (rxm_ep->sar.limit > RXM_SAR_LIMIT) {
+			rxm_ep->sar.limit = RXM_SAR_LIMIT;
```

Comment:
Do we need to limit this to RXM_SAR_LIMIT? Using msg_info->tx_attr->size helps keep the MSG tx queue full.

Reply:
I think having RXM_SAR_LIMIT is more understandable by user to control SAR that can be transferred by means of SAR protocol.